pandas/io/excel.py

```diff
@@ -1609,6 +1609,68 @@ def write_cells(self, cells, sheet_name=None, startrow=0, startcol=0,
                           startcol + cell.col,
                           val, style)
 
+    # Map from openpyxl-oriented styles to flatter xlsxwriter representation
```

Comment:
I think the code would be simpler to make this style formatting into a separate class (rather than have it live in functions sitting in the main excel code). can you refactor to make this cleaner.